core/src/prediction/collector/DefaultPredictionCollector.cpp

```diff
@@ -15,14 +15,17 @@
   along with grf. If not, see <http://www.gnu.org/licenses/>.
  #-------------------------------------------------------------------------------*/
 
+#include <future>
 #include <stdexcept>
 
 #include "prediction/collector/DefaultPredictionCollector.h"
+#include "commons/utility.h"
 
 namespace grf {
 
-DefaultPredictionCollector::DefaultPredictionCollector(std::unique_ptr<DefaultPredictionStrategy> strategy):
-    strategy(std::move(strategy)) {}
+DefaultPredictionCollector::DefaultPredictionCollector(std::unique_ptr<DefaultPredictionStrategy> strategy,
+                                                       uint num_threads):
+    strategy(std::move(strategy)), num_threads(num_threads) {}
 
 std::vector<Prediction> DefaultPredictionCollector::collect_predictions(
```

Comment:
This looks like the same logic as in `OptimizedPredictionCollector` -- we could consider moving this method into a shared superclass. To me it's not so much logic to be worth it yet, but is something to keep in mind.